### PR TITLE
Rvp rebase 1

### PR DESCRIPTION
**Pull request by Liaoshihua**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (linsinan1995) [inline .vscode/settings.json]: unnecessary file

File: gas/config/tc-riscv.c [hunk 2]
```diff
@@ -2894,6 +2953,97 @@ riscv_ip (char *str, struct riscv_cl_insn *ip, expressionS *imm_expr,
 		}
 	      break;
 
+case 'n':
```
Comment (linsinan1995) [inline gas/config/tc-riscv.c:2930]: formatting

File: bfd/elfxx-riscv.c [hunk 3]
```diff
@@ -1226,6 +1226,10 @@ static struct riscv_supported_ext riscv_supported_std_z_ext[] =
   {"zvl16384b",		ISA_SPEC_CLASS_DRAFT,		1, 0,  0 },
   {"zvl32768b",		ISA_SPEC_CLASS_DRAFT,		1, 0,  0 },
   {"zvl65536b",		ISA_SPEC_CLASS_DRAFT,		1, 0,  0 },
+
+  {"zbpbo",		ISA_SPEC_CLASS_DRAFT,		0, 9,  11 },
+  {"zpn",		  ISA_SPEC_CLASS_DRAFT,		0, 9,  11 },
+  {"zpsf",		ISA_SPEC_CLASS_DRAFT,		0, 9,  11 },
```
Comment (pz9115) [inline bfd/elfxx-riscv.c:1232]: Add additional space line.

File: gas/testsuite/gas/riscv/insn-dsp-zbpbo.d [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+#source: insn-dsp-zbpbo.s
+#objdump: -d
+
+.*:[ 	]+file format .*
+
+
+Disassembly of section .text:
+
+0+000 <dsp>:
+[ 	]+.*:[ 	]+.*[ 	]+clz[ 	]+a1,a2
```
Comment (pz9115) [inline gas/testsuite/gas/riscv/insn-dsp-zbpbo.d:11]: use the caculated opcdoe instead of *

File: gas/testsuite/gas/riscv/insn-dsp.d [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+#objdump: -d -M no-aliases
+
+
+.*:[ 	]+file format .*
+
+
+Disassembly of section .text:
+
+0+000 <dsp>:
+[ 	]+.*:[ 	]+.*[ 	]+add16[ 	]+a1,a2,a3
```
Comment (pz9115) [inline gas/testsuite/gas/riscv/insn-dsp.d:12]: need opcode

File: gas/testsuite/gas/riscv/insn-dsp64-zbpbo.d [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+#source: insn-dsp64-zbpbo.s
+#objdump: -d
+
+.*:[ 	]+file format .*
+
+
+Disassembly of section .text:
+
+0+000 <dsp64>:
+[ 	]+.*:[ 	]+.*[ 	]+cmix[ 	]+a1,a2,a3,a4
```
Comment (pz9115) [inline gas/testsuite/gas/riscv/insn-dsp64-zbpbo.d:11]: same, opcode required

File: gas/testsuite/gas/riscv/rvp_csr.s [hunk 7]
```diff
@@ -0,0 +1,6 @@
+rvp:
+    csrwi vxsat,0
+    csrci vxsat,1
+    csrr t0,vxsat
```
Comment (pz9115) [inline gas/testsuite/gas/riscv/rvp_csr.s:4]: Is the vxsat csr in RVV?

**Review by pz9115 — CHANGES_REQUESTED** [review]

File: gas/testsuite/gas/riscv/rvp_csr.s (hunk 7 above)
Comment (Liaoshihua) [reply gas/testsuite/gas/riscv/rvp_csr.s:4]: yes, the csr "vxsat" includes in RVV as well.

**Pull request closed without merging**